lib/multigrid.cpp

```diff
@@ -824,6 +779,23 @@ namespace quda
     popLevel();
   }
 
+  void MG::orthonormalize_vectors(const std::vector<ColorSpinorField*>& vecs, int count) const {
+    int num_vec = (count == -1) ? static_cast<int>(vecs.size()) : count;
+
+    if (num_vec > (int)vecs.size())
+      errorQuda("Trying to orthonormalize %d vectors which is larger than std::vector size %ld", num_vec, vecs.size());
+
+    for (int i = 0; i < num_vec; i++) {
```

Comment:
Why not do this using a CA-approach?  Even a two-pass algorithm would likely be much faster.

I'd suggest at least making it an option, anyway.

Reply:
One answer is "no good reason". My motivation at the time was it interferes with the `if (nrm2 > 1e-16)` check (since that's done before normalization), but we might be fine just throwing that out---I don't think I've ever hit it in practice, regardless of how good, bad, or stupid my setup was.

Reply:
Yeah I think that nrm check is a holdover from my original qcdmg code I wrote when I was a postdoc.  😜

Reply:
Punted because we need an in-place `cax_U`. Recorded for future reference in #1286